web/src/elements/decorators/intersection-observer.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file Intersection Observer Decorator for LitElement
+ */
```

Comment:
This decorator was originally designed to help with stale form issues in the upcoming dialog/modals fixes. The implementation can also handle our form viewport checks.